File: 1354-find-players-with-zero-or-one-losses/1354-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        HashMap<Integer, Integer> loss = new HashMap();
        for(int[] stat : matches){
            loss.put(stat[1], loss.getOrDefault(stat[1], 0) + 1);
            loss.putIfAbsent(stat[0], 0); // remember this
        }
        List<Integer> zeroLosses = new ArrayList<>();
        List<Integer> oneLoss = new ArrayList<>();

        for(Map.Entry<Integer, Integer> entry: loss.entrySet()){
            if (entry.getValue() == 0)
                zeroLosses.add(entry.getKey());
            if(entry.getValue() == 1)
                oneLoss.add(entry.getKey());
        }
        // Sort the lists
        zeroLosses.sort(null); //and this
        oneLoss.sort(null);

        // Return the result
        List<List<Integer>> result = new ArrayList<>();
        result.add(zeroLosses);
        result.add(oneLoss);

        return result;
    }
}

File: 1354-find-players-with-zero-or-one-losses/1354-find-players-with-zero-or-one-losses.rs
use std::collections::HashMap;
impl Solution {
    pub fn find_winners(matches: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let mut lossMap : HashMap<i32, i32> = HashMap::new();

        for i in matches{
            *lossMap.entry(i[1]).or_insert(0) += 1;
            if !lossMap.contains_key(&i[0]){
                lossMap.entry(i[0]).or_insert(0);
            }
        }

        let mut zero_losses = Vec::new();
        let mut one_loss = Vec::new();

        for (&key, &val) in lossMap.iter(){
            if val == 0{
                zero_losses.push(key);
            }
            if val == 1{
                one_loss.push(key);
            }
        }
        zero_losses.sort();
        one_loss.sort();

        vec![zero_losses, one_loss]
        
    }
}